src/devices/Ads1115/Ads1115.cs

```diff
@@ -14,12 +16,86 @@ namespace Iot.Device.Ads1115
+
+        private ComparatorPolarity _comparatorPolarity;
+        private ComparatorLatching _comparatorLatching;
+        private ComparatorQueue _comparatorQueue;
+
+        /// <summary>
+        /// Initialize a new Ads1115 device connected through I2C
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        /// <param name="inputMultiplexer">Input Multiplexer</param>
+        /// <param name="measuringRange">Programmable Gain Amplifier</param>
+        /// <param name="dataRate">Data Rate</param>
+        /// <param name="deviceMode">Initial device mode</param>
+        public Ads1115(I2cDevice i2cDevice, InputMultiplexer inputMultiplexer = InputMultiplexer.AIN0, MeasuringRange measuringRange = MeasuringRange.FS4096, DataRate dataRate = DataRate.SPS128,
+            DeviceMode deviceMode = DeviceMode.Continuous)
+        {
+            _i2cDevice = i2cDevice;
+            _inputMultiplexer = inputMultiplexer;
+            _measuringRange = measuringRange;
+            _dataRate = dataRate;
+            _gpioController = null;
+            _gpioInterruptPin = -1;
+            _deviceMode = deviceMode;
+            ComparatorMode = ComparatorMode.Traditional;
+            _comparatorPolarity = ComparatorPolarity.Low;
+            _comparatorLatching = ComparatorLatching.NonLatching;
+            _comparatorQueue = ComparatorQueue.Disable;
+            _shouldDispose = false;
+
+            SetConfig();
+            DisableAlertReadyPin();
+        }
+
+        /// <summary>
+        /// Initialize a new Ads1115 device connected through I2C with an additional GPIO controller for interrupt handling.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        /// <param name="gpioController">The GPIO Controller used for interrupt handling</param>
+        /// <param name="gpioInterruptPin">The pin number where the interrupt line is attached on the GPIO controller</param>
+        /// <param name="shouldDispose">True (the default) if the GPIO controller shall be disposed when disposing this instance</param>
+        /// <param name="inputMultiplexer">Input Multiplexer</param>
+        /// <param name="measuringRange">Programmable Gain Amplifier</param>
+        /// <param name="dataRate">Data Rate</param>
+        /// <param name="deviceMode">Initial device mode</param>
+        public Ads1115(I2cDevice i2cDevice,
+            GpioController gpioController, int gpioInterruptPin, bool shouldDispose = true, InputMultiplexer inputMultiplexer = InputMultiplexer.AIN0, MeasuringRange measuringRange = MeasuringRange.FS4096, DataRate dataRate = DataRate.SPS128,
+            DeviceMode deviceMode = DeviceMode.Continuous)
+            : this(i2cDevice, inputMultiplexer, measuringRange, dataRate, deviceMode)
+        {
+            _gpioController = gpioController ?? throw new ArgumentNullException(nameof(gpioController));
```

Comment:
Should we just initialize a board gpioController in this case and set shouldDispose to true?

Reply:
Since the controller argument is not optional here, I thought it would be better to explicitly require the user to pass a valid instance. I'm still not so much a fan of this "lets try to guess what the user wanted" approach. But I can change it if you think it would be useful.

Reply:
Ok, if it is not optional, I'm fine with the way it is.